src/Microsoft.AspNetCore.ResponseCaching/Internal/LoggerExtensions.cs

```diff
@@ -0,0 +1,322 @@
+                logLevel: LogLevel.Debug,
+                eventId: 21,
+                formatString: $"The content requested is not modified. The ETag {{ETag}} in the {HeaderNames.IfNoneMatch} header matched the ETag of a cached entry.");
+            _logIfUnmodifiedSinceSend304 = LoggerMessage.Define<DateTimeOffset, DateTimeOffset>(
+                logLevel: LogLevel.Debug,
+                eventId: 22,
+                formatString: $"The content requested is not modified. The last modified date of {{LastModified}} is before the date {{IfUnmodifiedSince}} specified in the {HeaderNames.IfUnmodifiedSince} header.");
+            _log304NotModifiedServed = LoggerMessage.Define(
+                logLevel: LogLevel.Information,
+                eventId: 23,
+                formatString: $"The content requested is not modified. A {StatusCodes.Status304NotModified} {ReasonPhrases.GetReasonPhrase(StatusCodes.Status304NotModified)} response will be served.");
+            _logCachedResponseServed = LoggerMessage.Define(
+                logLevel: LogLevel.Information,
+                eventId: 24,
+                formatString: "A valid cached response was retrieved for this request. The cached response will be served.");
+            _log504GatewayTimeoutServed = LoggerMessage.Define(
+                logLevel: LogLevel.Information,
+                eventId: 25,
+                formatString: $"No valid response was retrieved for this request but a only-if-cached cache directive was specified. A {StatusCodes.Status504GatewayTimeout} {ReasonPhrases.GetReasonPhrase(StatusCodes.Status504GatewayTimeout)} response will be served.");
+            _logNoResponseServed = LoggerMessage.Define(
+                logLevel: LogLevel.Information,
+                eventId: 26,
+                formatString: "No valid response was retrieved for this request.");
+            _logVaryByRulesStored = LoggerMessage.Define<string>(
+                logLevel: LogLevel.Information,
+                eventId: 27,
+                formatString: "Vary by rules were updated at key {Key}.");
+            _logResponseStored = LoggerMessage.Define<string>(
+                logLevel: LogLevel.Information,
+                eventId: 28,
+                formatString: "The response for this request was stored at key {Key}.");
+            _logResponseNotStored = LoggerMessage.Define(
+                logLevel: LogLevel.Information,
+                eventId: 29,
+                formatString: "The response could not be stored for this request.");
+            _logResponseContentLengthMismatchNotStored = LoggerMessage.Define(
+                logLevel: LogLevel.Information,
+                eventId: 30,
+                formatString: $"The response could not be stored for this request. The content length specified in the {HeaderNames.ContentLength} header did not match the length of the response.");
```

Comment:
Warning? Error? This is an app error, though not a cache error. "The response could not be cached because the Content-Length did not match the body length."